lib/new_relic/agent/instrumentation/roda/roda_transaction_namer.rb

```diff
@@ -0,0 +1,38 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+module NewRelic
+  module Agent
+    module Instrumentation
+      module Roda
+        module TransactionNamer
+          extend self
+
+          def initial_transaction_name(request)
+            transaction_name(::NewRelic::Agent::UNKNOWN_METRIC, request)
+          end
+
+          ROOT = '/'.freeze
+
+          def transaction_name(path, request)
+            verb = http_verb(request)
+            path = request.path if request.path
+            name = path.gsub(%r{^[/^\\A]*(.*?)[/\$\?\\z]*$}, '\1') # remove any rouge slashes
```

Comment:
1. I think if you're going to the trouble of freezing something as simple as the `ROOT` slash, you should freeze this regex pattern here and have it live outside the method as well to avoid redefining it every time the method gets called.
2. Perhaps consider removing slashes of other colors as well...

Reply:
[Froze the regex](https://github.com/newrelic/newrelic-ruby-agent/pull/2144/files#diff-b4632c237fe6e6069aa8e3bf333b378bb34b6c86f63df1ad4c1b24d198fd7887R17). Ah shoot, I didn't mean to leave all the other colors out—how embarrassing. Now we get to capture all rogue slashes, not just rouge.